### PR TITLE
Claude corrections 220419

### PR DESCRIPTION
**Pull request by nchauvat**

### CODE REVIEW

File: public/vocables/js/sparqlProxies/sparql_OWL.js [hunk 1]
```diff
@@ -232,7 +232,8 @@ var Sparql_OWL = (function () {
             "  WHERE {{";
 
         //   query += "?concept rdf:type ?type. "
-        query += "?concept rdf:type* owl:Class. ";
+        //query += "?concept rdf:type* owl:Class. ";
```
Comment (nchauvat) [inline public/vocables/js/sparqlProxies/sparql_OWL.js:235]: could you delete the two commented lines ? (starting with //)

File: public/vocables/js/search/searchUtil.js [hunk 2]
```diff
@@ -335,16 +335,17 @@ var SearchUtil = (function () {
...
+           // indexSource: 1,
             indexName: indexName,
             data: JSON.stringify(data),
             options: JSON.stringify(options),
         };
 
         $.ajax({
             type: "POST",
-            url: Config.serverUrl,
+            urlX: Config.serverUrl,
```
Comment (nchauvat) [inline public/vocables/js/search/searchUtil.js:347]: I do not think urlX is a parameter of ajax(). let us remove it.

File: public/vocables/js/search/searchUtil.js [hunk 3]
```diff
@@ -335,16 +335,17 @@ var SearchUtil = (function () {
         //  MainController.UI.message("indexing " + data.length)
         var options = {replaceIndex: replaceIndex, owlType: "Class"};
         var payload = {
-            // dictionaries_indexSource: 1,
-            indexSource: 1,
+           // dictionaries_indexSource: 1,
+           // indexSource: 1,
```
Comment (nchauvat) [inline public/vocables/js/search/searchUtil.js:339]: are these comments necessary ? if not, let's delete them.

File: bin/KG/dictionariesManager..js [hunk 4]
```diff
@@ -47,6 +47,8 @@ var DictionariesManager = {
     },
 
     indexSource: function (indexName, data, _options, callback) {
+        if (typeof data === "string") data = JSON.parse(data);
+        if (typeof _options === "string") _options = JSON.parse(_options);
```
Comment (nchauvat) [inline bin/KG/dictionariesManager..js:51]: _options was prefixed with an underscore to make eslint happy when _options was not used. Now that _options is used, let's name it options without a _ prefix.

File: bin/KG/dictionariesManager..js [hunk 5]
```diff
@@ -47,6 +47,8 @@ var DictionariesManager = {
     },
 
     indexSource: function (indexName, data, _options, callback) {
+        if (typeof data === "string") data = JSON.parse(data);
```
Comment (nchauvat) [inline bin/KG/dictionariesManager..js:50]: the data is not expected to be a string, so let's throw an exception if that is not the case, but let's not convert on the fly to accept multiple types.

File: public/vocables/js/search/elasticSearchProxy.js [hunk 6]
```diff
@@ -4,6 +4,9 @@ var ElasticSearchProxy = (function () {
 
     self.queryElastic = function (query, indexes, callback) {
         if (!indexes) indexes = context.curentSearchIndexes;
+        if (!Array.isArray(indexes)) {
+            indexes = [indexes];
+        }
```
Comment (nchauvat) [inline public/vocables/js/search/elasticSearchProxy.js:9]: if the idea is to make sure indexes is an Array, let's instead throw an exception if that is not the case.

File: public/vocables/js/search/elasticSearchProxy.js [hunk 7]
```diff
@@ -14,6 +17,7 @@ var ElasticSearchProxy = (function () {
             indexes: indexes,
         };
 
+        var x = JSON.stringify(payload);
```
Comment (nchauvat) [inline public/vocables/js/search/elasticSearchProxy.js:20]: is this needed ? if not, let's remove it.

File: public/vocables/js/search/searchUtil.js [hunk 8]
```diff
@@ -331,9 +333,10 @@ var SearchUtil = (function () {
     self.indexData = function (indexName, data, replaceIndex, callback) {
         if (data.length == 0) return callback();
         //  MainController.UI.message("indexing " + data.length)
-        var options = { replaceIndex: replaceIndex, owlType: "Class" };
+        var options = {replaceIndex: replaceIndex, owlType: "Class"};
         var payload = {
-            dictionaries_indexSource: 1,
+            // dictionaries_indexSource: 1,
+            indexSource: 1,
```
Comment (nchauvat) [inline public/vocables/js/search/searchUtil.js]: can we both use indexSource and indexName in the same query ?

File: public/vocables/js/search/searchUtil.js [hunk 9]
```diff
@@ -185,7 +185,7 @@ var SearchUtil = (function () {
         if (ids) {
             size = ids.length + 10;
             var str = "";
-            var header = { index: index };
+            var header = {index: index};
```
Comment (nchauvat) [inline public/vocables/js/search/searchUtil.js:188]: is this change needed ?

same question for all the whitespace changes below.

File: public/vocables/js/search/searchUtil.js [hunk 10]
```diff
@@ -37,7 +37,12 @@ var SearchUtil = (function () {
                                 if (err) return callbackSeries(err);
 
                                 indexes = [];
-                                if (toSources && !Array.isArray(toSources)) toSources = [toSources];
+                                if (toSources) {
+                                    if (!Array.isArray(toSources)) toSources = [toSources];
```
Comment (nchauvat) [inline public/vocables/js/search/searchUtil.js:41]: if toSources has to be an array, let's raise an exception when it is not.

File: public/vocables/js/search/searchUtil.js [hunk 11]
```diff
@@ -37,7 +37,12 @@ var SearchUtil = (function () {
...
+                                    toSources.forEach(function (_source) {
+                                        indexes.push(_source.toLowerCase());
+                                    });
+                                }
```
Comment (nchauvat) [inline public/vocables/js/search/searchUtil.js:45]: should we make indexes a Set ?

would it work to write the following ?

indexes = toSources.map((name) => name.toLowerCase())

**Comment by YoelisA** [comment]
Looks good to me I'll merge it!
